tsds-grafana-handler/tsds_grafana_handler.py

```diff
@@ -37,17 +37,55 @@ def get_timeframe(seconds):
 
         return "{0}s".format(seconds)
 
-def findtarget_names(tsds_result, alias_list, target_aliases):
+def is_metric(key):
+        ''' Returns true if key is a metric
         '''
+
+        if 'values.' in key:
+                return False
+        else:
+                return True
+
+def metric_label(tsds_result):
+        ''' Builds a graph label based on a tsds result
+
+        tsds_result tsds result containing datapoints and metric names
+        '''
+        # Get all metric names to populate as the target name.
+        targetname = []
+        nonValues  = []
+        for k, v in tsds_result.iteritems():
+                if is_metric(k):
+                        nonValues.append(k)
+        nonValues.sort(key=natural_keys)
+
+        # May need to handle utf8 conversions here?
+        for nV in nonValues:
+                if nV in tsds_result: targetname.append(tsds_result[nV])
+
+        return targetname
+
+def findtarget_names(tsds_result, alias_list, target_aliases):
+        ''' Builds a list of datapoint set names and their graph label
+
         tsds_result tsds result containing datapoints and metric names
         alias_list List of template variables to be populated
         target_aliases Hash of datapoint names to datapoint name aliases
+
+        ex. aggregate(values.input, 300, average) => Input (5m averages)
         '''
         returnname = []
 
         for key, value in tsds_result.iteritems():
                 # Ignores keys describing requested metrics
-                if not isinstance(value, list):
+                if is_metric(key):
+                        continue
+
+                datapoint_args = map(lambda x: x.strip(), re.split('[(,)]', key))
+
+                # [u'sum', u'aggregate', u'values.output', u'300', u'average', u'', u'']
+                if not datapoint_args[0] == u'aggregate':
```

Comment:
I'm prefer not testing against unicode versions. The str comp in python I believe will treat ascii and non-ascii strings of ascii only chars the same, and since we're testing against "aggregate" here it's safe.

>>> 'aggregate' == u'aggregate'
True